analyzer/optane/check_dimm_software_trigger_counter

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+#set -x
+
+# Prevent repeated sourcing
+if [ -n "${_MODULE_CHECK_DIMM_SOFTWARE_TRIGGER_COUNTER_}" ]; then
+  return
+fi
+_MODULE_CHECK_SOFTWARE_TRIGGER_COUNTER__="Loaded"
+
+# Validate DIMM Software Trigger is not equal to 0
+function optane_check_dimm_software_trigger_counter() {
+  local FNAME=$1        # File name to process
+  local ERR_STATE=false # Used for error reporting
+
+  # Exit the function with INFO message if the file to process doesn't exist
+  if [[ ! -f ${FNAME} ]] ; then
+    info_msg " Could not process '${FNAME}'. File not found"
+    REPORT_COUNT_INFO=$((REPORT_COUNT_INFO+1)) # Increment the INFO count
+    return
+  fi
+
+
+  # Get the DimmID, Property, and Value fields from the input file
+  while IFS='|' read -r DimmID Property Value 
+  do
+    DimmID="$(echo ${DimmID})"         # Remove leading and trailing whitespaces
+    Property="$(echo ${Property})" # Remove leading and trailing whitespaces
+    PropertyValue="$(echo ${Value})"   # Remove leading and trailing whitespaces
+    if [ ${OPT_VERBOSITY} -ge 3 ]; then
+      debug_msg "DimmID ${DimmID}: ${Property}=${Value}"
+    fi
+    # Report PMem Modules that have Software Trigger Counter greater than 0
+    if [[ "${PropertyValue}" -ne 0 ]]; then
+      ERR_STATE=true
+    fi
+  done <<<  "$(${GREP} SoftwareTriggerCounter ${FNAME})" || return 0 
+
+  # check for the final pass or fail
+  if [ "${ERR_STATE}" = true ] ; then
+    echo "${STR_FAIL} ${FUNCNAME[0]} : One or more PMem modules reported Software Trigger Counter equal to 0" 
```

Comment:
The string is for the PASS condition (SoftwareTriggerCounter=0), not FAIL (this IF condition). 
Suggest using "One or more PMem modules reported SoftwareTriggerCounter > 0"